lib/backup_restore/backuper.rb

```diff
@@ -452,6 +456,13 @@ def save_log(message, timestamp)
       @logs << "[#{timestamp}] #{message}"
     end
 
+    def before_run
+      if @with_uploads && BackupRestore.strict_backup_creation?
+        Discourse.enable_readonly_mode(Discourse::USER_READONLY_MODE_KEY)
+        SiteSetting.include_s3_uploads_in_backups = true
```

Comment:
Is this site setting the only way to trigger this? I would prefer if it could be passed as an option rather than setting a global setting like this.